doc/manual/src/development/debugging.md

```diff
@@ -0,0 +1,61 @@
+# Debugging Nix
+
+This section provides instructions on how to build and debug Nix with debug
+symbols enabled. It assumes you are using Nix with the [`flakes`] and
+[`nix-command`] experimental features enabled.
+
+[`flakes`]: @docroot@/development/experimental-features.md#xp-feature-flakes
+[`nix-command`]: @docroot@/development/experimental-features.md#xp-nix-command
+
+## Building Nix with Debug Symbols
+
+First, ensure you have set up the development environment as described in the
+[building documentation](./building.md).
+
+In the development shell, set the `mesonBuildType` environment variable to
+`debug` before configuring the build:
+
+```console
+[nix-shell]$ export mesonBuildType=debug
```

Comment:
Perhaps this is needed for the Meson build, but our regular build has debug symbols by default.

Reply:
Why do you mean by regular build?

Reply:
@edolstra is talking about the old autotools buildsystem that we are migrating away from. Do we want debugoptimized instead of debug here?

```suggestion
[nix-shell]$ export mesonBuildType=debugoptimized
```

Reply:
Optimized makes sense to me if the debugger is this able to nicely link back into the source code.